### PR TITLE
api: refuse to create a device if tenant is invadid

### PR DESCRIPTION
**Pull request by eduardoveiga**

### CODE REVIEW

**Review by otavio — APPROVED** [review]

**Review by otavio — CHANGES_REQUESTED** [review]
This requires a migration to remove devices in this situation. We may have leftovers and it is important to remove them here.

File: api/authsvc/service.go [hunk 1]
```diff
@@ -65,6 +65,11 @@ func (s *service) AuthDevice(ctx context.Context, req *models.DeviceAuthRequest)
 		LastSeen:  time.Now(),
 	}
 
+	namespace, err := s.store.NamespaceGet(ctx, device.TenantID)
```
Comment (otavio) [inline api/authsvc/service.go:68]: Please add following comment here:
```go
# The order here is critical as we don't want to register devices if the tenant id is invalid
```

**Review by otavio — COMMENTED** [review]
Please add the migration in this PR so we review it  together.

File: api/store/mongo/migrations/main.go [hunk 2]
```diff
@@ -33,6 +33,8 @@ func GenerateMigrations() []migrate.Migration {
 		migration_21,
 		migration_22,
 		migration_23,
+		migration_24,
```
Comment (otavio) [inline api/store/mongo/migrations/main.go:36]: here is wrong. This should include just 25.

**Review by gustavosbarreto — APPROVED** [review]

File: api/store/mongo/migrations/main.go [hunk 3]
```diff
@@ -34,6 +34,7 @@ func GenerateMigrations() []migrate.Migration {
 		migration_22,
 		migration_23,
 		migration_24,
+		migration_25,
```
Comment (otavio) [inline api/store/mongo/migrations/main.go:37]: This belongs to the migration patch.

**Review by otavio — CHANGES_REQUESTED** [review]

**Review by gustavosbarreto — CHANGES_REQUESTED** [review]
Why didn't you remove the devices in migration?

**Pull request merged**